### ISSUE TITLE
Can TextureFilter provide default texture coordinates?

### ISSUE THREAD

**Issue by djhoese** [vispy/vispy#2427]
I'm helping a user with this stackoverflow question:

https://stackoverflow.com/questions/74300215/vispy-surface-plot-highlight-particular-section-with-color-and-image/74348810

Where they want to adapt their SurfacePlot usage to have a texture from an image that they are loading, following this example:

https://vispy.org/gallery/scene/mesh_texture.html

The user wasn't really sure what they should pass for `texcoords` when they create a TextureFilter. My question for the other vispy maintainers is: is it possible to provide a default for these coordinates?

As I started typing this I thought the texture coordinates were on the same shape as the texture/image being applied, but they are the same shape as the Mesh vertices array aren't they? At the very least we should probably be more specific in this docstring:

https://github.com/vispy/vispy/blob/b9c3ee5888466d6b0c0f483952bb620a1e79233a/vispy/visuals/filters/mesh.py#L27-L28

**Comment by djhoese** [comment]
Maybe this is possible for the SurfacePlot specifically since we know there are `z.shape[0] * z.shape[1]` vertices. We could use that shape information to map the image equally over the SurfacePlot or use the `x` and `y` values of the surface plot to match the distribution of those variables. Thoughts?

**Comment by asnt** [comment]
> Maybe this is possible for the SurfacePlot specifically since we know there are `z.shape[0] * z.shape[1]` vertices. We could use that shape information to map the image equally over the SurfacePlot or use the `x` and `y` values of the surface plot to match the distribution of those variables. Thoughts?

I would agree with your reasoning, @djhoese. For an arbitrary mesh it's not obvious what the mapping between the 3D vertices and the 2D plane of the texture image should be. This might be specific to the application. But for meshes with standard shapes (2D grid as in the surface plot, sphere, cube, tetrahedron...), there exist one or several standard ways of mapping a texture that we might want to offer as default solutions.
